PssNationalInstance/src/main/java/com/intellisoft/pssnationalinstance/service_impl/impl/SurveysServiceImpl.java

```diff
@@ -268,6 +268,79 @@ public Results updateSurvey(String surveyId) {
         return new Results(400, "Resource not found");
     }
 
+    @Override
+    public Results listAllSurveys(String status) {
+        if (status == null || status.isEmpty()) {
+            status = "ALL";
+        }
+
+        Iterable<Surveys> surveysList = surveysRepo.findAll();
+        List<DbSurveyDetails> dbSurveyDetailsList = new ArrayList<>();
+
+        for (Surveys survey : surveysList) {
```

Comment:
There is a repeated code segment between `listAllSurveys` and `listAdminSurveys` methods. You can create a common method that is referenced by the 2 methods to avoid the code duplication.